titan-compiler/symtab.lua

```diff
@@ -0,0 +1,35 @@
+local symtab = {}
+
+symtab.__index = symtab
+
+function symtab.new()
+    return setmetatable({ blocks = {} }, symtab)
+end
+
+function symtab:open_block()
+    table.insert(self.blocks, {})
+end
+
+function symtab:close_block()
+    assert(#self.blocks > 0, "stack underflow")
+    table.remove(self.blocks)
+end
+
```

Comment:
I love this style of wrapper functions because it states intent in the code so clearly. Can't wait until the day comes and we have aggressive inlining in Titan so I can use this freely without Lua people nagging me with "but you're creating a closure every time!"

Reply:
I like the idea, but this make the error treatment awkward in the check module:
```
local ok, err = st:with_blocks(function() 
    local ok, err = node:foreach(bindvars, st)
    if not ok then return false, err end
end)
if not ok then return false, err end
```

Reply:
In fact, I'm going to remove the `if not ok ...` and implement this.